tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Core.cs

```diff
@@ -58,6 +59,18 @@ private static void GetCookieKeyValueFromIndex(IRequestCookieCollection cookies,
         value = cookie.Value;
     }
 
+    [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
I assume this is to protect the `CoreHttpContextStore`, but is it really necessary? 🤔 If it's potentially going to crash, I would expect to see a try-catch in here...

Reply:
Seems it's necessary as if it's inlined it will try to load `Microsoft.AspNetCore.Hosting.Abstractions` and crashes. I dont think we can try/catch it as it's assembly loading the moment this method is being called.. (?)  
![image](https://github.com/user-attachments/assets/b0ed306d-8e7a-4589-a3e5-0089c2d631c0)

We had the discussion [here ](https://dd.slack.com/archives/CC7Q58YRG/p1730732361004009?thread_ts=1730731618.767639&cid=CC7Q58YRG)